src/Microsoft.DocAsCode.Build.Engine/Incrementals/DependencyGraph.cs

```diff
@@ -3,61 +3,68 @@
+            { DependencyTypeName.Uid, new DependencyType { Name = DependencyTypeName.Uid, IsTransitive = false, TriggerBuild = false } },
+        };
 
-        public DependencyGraph()
-            : this(new HashSet<DependencyItem>())
+        internal DependencyGraph()
+            : this(new HashSet<DependencyItem>(), _defaultTypes.ToDictionary(p => p.Key, p => p.Value))
         {
         }
 
-        private DependencyGraph(HashSet<DependencyItem> dependencies)
+        private DependencyGraph(HashSet<DependencyItem> dependencies, Dictionary<string, DependencyType> types)
         {
             _dependencyItems = dependencies;
+            _types = types;
             RebuildIndex();
         }
 
-        static DependencyGraph()
+        public IReadOnlyDictionary<string, DependencyType> DependencyTypes
         {
-            // Register default dependency types
-            RegisterDependencyType(new DependencyType
-            {
-                Name = DependencyTypeName.Include,
-                IsTransitive = true,
-                TriggerBuild = true,
-            });
-            RegisterDependencyType(new DependencyType
-            {
-                Name = DependencyTypeName.Uid,
-                IsTransitive = false,
-                TriggerBuild = false,
-            });
+            get { return _types; }
         }
 
-        public static IReadOnlyDictionary<string, DependencyType> DependencyTypes
+        public void RegisterDependencyType(DependencyType dt)
         {
-            get { return _types; }
+            RegisterDependencyType(new List<DependencyType> { dt });
         }
 
-        public static void RegisterDependencyType(DependencyType dt)
+        public void RegisterDependencyType(IEnumerable<DependencyType> dts)
         {
-            DependencyType stored;
-            if (_types.TryGetValue(dt.Name, out stored))
+            if (dts == null)
             {
-                Logger.LogWarning($"Dependency type {JsonUtility.Serialize(dt)} isn't registered successfully because a type with name {dt.Name} is already registered. Already registered one: {JsonUtility.Serialize(stored)}.");
-                return;
+                throw new ArgumentNullException(nameof(dts));
+            }
+            foreach (var dt in dts)
+            {
+                DependencyType stored;
+                if (_types.TryGetValue(dt.Name, out stored))
+                {
+                    if (stored.TriggerBuild != dt.TriggerBuild || stored.IsTransitive != dt.IsTransitive)
+                    {
+                        Logger.LogError($"Dependency type {JsonUtility.Serialize(dt)} isn't registered successfully because a different type with name {dt.Name} is already registered. Already registered one: {JsonUtility.Serialize(stored)}.");
+                        throw new InvalidDataException($"A different dependency type with name {dt.Name} is already registered");
+                    }
+                    continue;
```

Comment:
>continue; [](start = 20, length = 9)

print a verbose that the dependency type is same with earlier value and ignored?